File: backend/app/models/weather.py
from datetime import datetime

from sqlalchemy import Column, Integer, String, Float, Boolean, DateTime

from app.core.database import Base


class WeatherData(Base):

    __tablename__ = "weather_data"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    site_name = Column(
        String,
        nullable=False
    )

    temperature = Column(
        Float,
        nullable=False
    )

    humidity = Column(
        Float,
        nullable=False
    )

    wind_speed = Column(
        Float,
        nullable=False
    )

    rainfall = Column(
        Float,
        nullable=False,
        default=0.0
    )

    weather_condition = Column(
        String,
        nullable=True
    )

    weather_description = Column(
        String,
        nullable=True
    )

    storm = Column(
        Boolean,
        default=False
    )

    weather_risk = Column(
        String,
        nullable=False
    )

    recorded_at = Column(
        DateTime,
        default=datetime.utcnow,
        nullable=False
    )

File: frontend/src/types/weather.ts
export type RiskLevel = "LOW" | "MEDIUM" | "HIGH" | "SEVERE" | string;

export interface Weather {
  id: number;
  site_name: string;

  temperature: number;
  humidity: number;
  wind_speed: number;
  rainfall: number;

  weather_condition: string;
  weather_description: string;

  storm: boolean;
  weather_risk: RiskLevel;

  timestamp?: string | null;
  recorded_at?: string | null;
}

export interface WeatherCurrentResponse {
  location: string;
  weather: Weather;
}

export interface WeatherHistoryResponse {
  records: Weather[];
}

export type WeatherHistory = Weather[];